test_cases/stdlib/check_unittest.py

```diff
@@ -94,13 +94,20 @@ def __gt__(self, other: Bacon) -> bool:
 ###
 
 
-@patch("sys.exit", new=Mock())
-def f(i: int) -> str:
+@patch("sys.exit")
+def f_default_new(i: int, mock: MagicMock) -> str:
+    return "asdf"
+
+
+@patch("sys.exit", new=42)
+def f_explicit_new(i: int) -> str:
     return "asdf"
 
 
-assert_type(f(1), str)
-f("a")  # type: ignore
+assert_type(f_default_new(1), str)
+f_default_new("a")  # Not an error due to ParamSpec limitations
+assert_type(f_explicit_new(1), str)
+f_explicit_new("a")  # type: ignore[arg-type]
```

Comment:
Nit: we currently don't use mypy error codes for `type: ignore`s in the `test_cases` directory: (see rationale in #8083)

```suggestion
f_explicit_new("a")  # type: ignore
```